pkg/query/storeset.go

```diff
@@ -263,6 +313,15 @@ func (s *StoreSet) getHealthyStores(ctx context.Context) map[string]*storeRef {
 				labels, minTime, maxTime, err := spec.Metadata(ctx, st.StoreClient)
 				if err != nil {
 					// Peer unhealthy. Do not include in healthy stores.
+					s.updateStoreStatus(&StoreStatus{
```

Comment:
if `err != nil`, labels, minTime, ... will be set to nil, zero, etc. I think you should have function `s.updateStoreStatusErr(addr, err error)` that handles the error case. It will make the code more compact and DRY. Also the minTime/maxTime will be uniform in case of all error conditions.